### DISCUSSION TITLE
Adding a 'C++' topic for a repository ...

### DISCUSSION THREAD

**Discussion by antoinecarme** [community/community#103006] [Repositories]
Hi all,

I am trying to add 'C++' as a topic for a repository main page. This does not seem to be allowed, '+' is too special. For the moment , I use 'cplusplus' ;)

https://github.com/mllite/TinyML_On_MicroControllers_Demo

Up to 20 topics can be displayed on the right of the main page.

Any help or work-around is welcome.

Thanks in advance

Antoine

**Comment by gil-son** [comment]
It's common, because some symbols are reserved, like: #, *, +

The same happen with C# = csharp

Then the tag cplusplus represent C++

**Comment by ghost** [comment]
Unfortunately, GitHub doesn't allow special characters like '+' in topics. Using "C-plus-plus" or "Cplusplus" are common workarounds. If you prefer not to use those alternatives, you might consider using "cpp" or "cplusplus" to represent C++.

I guess 'cpp' would be better

**Reply by ghost** [reply]
Actually I Don't know c++ so I tried it in topics section and it is actually a little frustrating when you want to write cplusplus

**Reply by ghost** [reply]
![Screenshot_2024-02-02-22-59-09-248_com android chrome](https://github.com/community/community/assets/121353576/b609abf2-4177-476d-8713-f9c4c0b67757)


As you can see many popular repositories had used these ways to include c++ in topics

**Comment by antoinecarme** [comment]
Thanks a lot guys. 

I will stick with cpp or cplusplus. 

This seems to be a hashtag syntax limitation. The same occurs on twitter and like.

https://www.hashtags.org/featured/what-characters-can-a-hashtag-include/

It is a good thing Github allows using hashtags internally to search repositories using topics. And 'cplusplus' refers indeed to C++

https://github.com/search?q=%23cplusplus&type=repositories

Cheers

Antoine

**Reply by ghost** [reply]
Thats good then close the discussions, that would be ideal right.....


Also Happy C-plus-plus Journey 😂🤣
